apis/network/v1alpha3/types.go

```diff
@@ -230,12 +230,11 @@ type SubnetList struct {
 // A PublicIPAddressSpec defines the desired state of a PublicIPAddress.
 type PublicIPAddressSpec struct {
 	xpv1.ResourceSpec `json:",inline"`
-	// +kubebuilder:validation:Required
-	ForProvider PublicIPAddressFormat `json:"properties"`
+	ForProvider       PublicIPAddressProperties `json:"forProvider"`
 }
 
-// PublicIPAddressFormat defines properties of the PublicIPAddress.
-type PublicIPAddressFormat struct {
+// PublicIPAddressProperties defines properties of the PublicIPAddress.
```

Comment:
FYI, looks like there are [many more](https://docs.microsoft.com/tr-tr/rest/api/virtualnetwork/public-ip-addresses/create-or-update) properties that can be specified for `PublicIPAdress`.

Reply:
We may consider adding them in later PRs unless there is immediate need, what do you think @muvaf?

Reply:
I personally prefer making it complete while I have the context of the resource, otherwise I'd have to load all that later again, but it can be merged as an alpha resource with less properties than full.

Reply:
It turns out that we are not sure regarding the immediate requirements around this resource. Hence, I've extended the API surface.